sfge/src/main/java/com/salesforce/cli/CacheCreator.java

```diff
@@ -0,0 +1,112 @@
+package com.salesforce.cli;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+
+import com.salesforce.config.SfgeConfig;
+import com.salesforce.config.SfgeConfigProvider;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/** Creates cache data that can be used to execute Delta runs. */
+public final class CacheCreator {
+    private static final Logger LOGGER = LogManager.getLogger(CacheCreator.class);
+
+    private Dependencies dependencies;
+
+    public CacheCreator() {
+        this.dependencies = new Dependencies();
+    }
+
+    CacheCreator(Dependencies dependencies) {
+        this.dependencies = dependencies;
+    }
+
+    /**
+     * Creates cache data that can be used for future delta runs
+     *
+     * @param result from the current execution
+     */
+    public void create(Result result) {
+        createFilesToEntriesCache(result.getFilesToEntriesMap());
+    }
+
+    private void createFilesToEntriesCache(FilesToEntriesMap filesToEntriesMap) {
+        final String jsonData = filesToEntriesMap.toJsonString();
+        final String filesToEntriesDataFilename = getFilesToEntriesDataFilename();
+
+        if (LOGGER.isInfoEnabled()) {
```

Comment:
Is this the standard way to log in SFGE? Doesn't sound optimal at all. Shouldn't the logging framework do it for us?